### PR TITLE
Allow users to to preview password fields in credential forms

### LINKED ISSUE #2584: Allow users to preview secrets and password when the creating/updating credentials
On the add credentials modal, we should allow users to preview passwords and secrets . 

Image

### PR DESCRIPTION
**Pull request by midigofrank**
### Description

This PR allows users to to preview password fields in credential forms. This feature already existed in the `NewInputs.input/1` function component. 

Closes #2584 

### Validation steps

1. Try creating/updating a credential having a password field, like the `http` adaptor credential.
2. You'll see that it now have the button to preview the password 

### Additional notes for the reviewer

There is slight problem with the existing implementation for preview. All is does is change the input type to `text`. 
For liveview forms with `phx-change`, the input gets restored back to a password type. It can be a bit annoying to the user

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [ ] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [x] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

### Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

### CODE REVIEW

File: lib/lightning_web/live/credential_live/json_schema_body_component.ex [hunk 1]
```diff
@@ -83,12 +83,12 @@ defmodule LightningWeb.CredentialLive.JsonSchemaBodyComponent do
...
-        %{"type" => "string"} -> :text_input
-        %{"type" => "integer"} -> :text_input
-        %{"type" => "boolean"} -> :checkbox
-        %{"anyOf" => [%{"type" => "string"}, %{"type" => "null"}]} -> :text_input
+        %{"format" => "uri"} -> "url"
+        %{"type" => "string", "writeOnly" => true} -> "password"
+        %{"type" => "string"} -> "text"
+        %{"type" => "integer"} -> "text"
+        %{"type" => "boolean"} -> "checkbox"
+        %{"anyOf" => [%{"type" => "string"}, %{"type" => "null"}]} -> "text"
```
Comment (stuartc) [inline lib/lightning_web/live/credential_live/json_schema_body_component.ex:91]: Curious, why were these changed to strings?

File: lib/lightning_web/live/credential_live/json_schema_body_component.ex [hunk 2]
```diff
@@ -123,15 +123,7 @@ defmodule LightningWeb.CredentialLive.JsonSchemaBodyComponent do
...
-        <%= apply(Phoenix.HTML.Form, @type, [
-          @form,
-          @field,
-          [
-            value: @value || "",
-            class: ~w(mt-1 focus:ring-primary-500 focus:border-primary-500 block
-               w-full shadow-sm sm:text-sm border-secondary-300 rounded-md)
-          ]
-        ]) %>
+        <.input type={@type} field={@form[@field]} value={@value || ""} />
```
Comment (stuartc) [inline lib/lightning_web/live/credential_live/json_schema_body_component.ex:126]: Nice! Is it possible to instead leverage the input components existing behaviour where it picks the value off the form field?

File: lib/lightning_web/live/credential_live/json_schema_body_component.ex (hunk 1 above)
Comment (midigofrank) [reply lib/lightning_web/live/credential_live/json_schema_body_component.ex:91]: Because `NewInputs.input/1` expects a text on the `type` assign

File: lib/lightning_web/live/credential_live/json_schema_body_component.ex (hunk 2 above)
Comment (midigofrank) [reply lib/lightning_web/live/credential_live/json_schema_body_component.ex:126]: Yeah, I tried that but for some reason, during editing it doesn't pick up that value
